api/src/modules/scenarios-features/scenario-features.service.ts

```diff
@@ -0,0 +1,135 @@
+import { In, Repository, SelectQueryBuilder } from 'typeorm';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { FiltersSpecification } from 'nestjs-base-service';
+import {
+  AppBaseService,
+  JSONAPISerializerConfig,
+} from '../../utils/app-base.service';
+
+import { AppInfoDTO } from '../../dto/info.dto';
+
+import { remoteConnectionName } from './entities/remote-connection-name';
+import { GeoFeature } from '../geo-features/geo-feature.api.entity';
+import { RemoteScenarioFeaturesData } from './entities/remote-scenario-features-data.geo.entity';
+import { RemoteFeaturesData } from './entities/remote-features-data.geo.entity';
+
+@Injectable()
+export class ScenarioFeaturesService extends AppBaseService<
+  RemoteScenarioFeaturesData,
+  never,
+  never,
+  AppInfoDTO
+> {
+  constructor(
+    @InjectRepository(GeoFeature)
+    private readonly features: Repository<GeoFeature>,
+    @InjectRepository(RemoteFeaturesData, remoteConnectionName)
+    private readonly remoteFeature: Repository<RemoteFeaturesData>,
+    @InjectRepository(RemoteScenarioFeaturesData, remoteConnectionName)
+    private readonly remoteScenarioFeatures: Repository<RemoteScenarioFeaturesData>,
+  ) {
+    super(remoteScenarioFeatures, 'scenario_features', 'scenario_feature');
+  }
+
+  setFilters(
+    query: SelectQueryBuilder<RemoteScenarioFeaturesData>,
+    filters?: FiltersSpecification['filter'],
+  ): SelectQueryBuilder<RemoteScenarioFeaturesData> {
+    if (filters?.scenarioId) {
+      return query.andWhere(`${this.alias}.scenario_id = :scenarioId`, {
+        scenarioId: filters?.scenarioId,
+      });
+    }
```

Comment:
one more comment about this I have just realised - this is totally undocumented (not least because BaseService is something I had to heavily reverse-engineer while also refactoring it and so on), but in theory, if we were to use this service from a different controller handler where we do collect fetchSpecification data from the request, and we pass the fetchSpecification on to service.findAll(), an API consumer could cause this filter to be added by using something like `?filter[scenarioId]=<value>`.

To avoid this, we could use the `info` param of `findAll()`, e.g. (in a controller) `this.service.findAll(fetchSpecification, { params: { scenarioId: <uuid> }});` - or something like this. We have instances where we hijack the fetchSpecification.filter prop to add our own filters like in the first controller in this module, but after Alex asked me about the info part of BaseService, which I had just ported over from the implementation I had inherited in an older project, I eventually realised what it may have been meant for 😁

Reply:
updated in https://github.com/Vizzuality/marxan-cloud/pull/139/commits/da722293b21e133072125a58b43f7aa94b4ef077